Sources/OAuth2Client/NXOAuth2Connection.m

```diff
@@ -418,7 +418,7 @@ - (void)connection:(NSURLConnection *)connection didReceiveResponse:(NSURLRespon
     if (self.statusCode == 401
         && client.accessToken.refreshToken != nil
         && authenticateHeader
-        && [authenticateHeader rangeOfString:@"expired_token"].location != NSNotFound) {
+        && ([authenticateHeader rangeOfString:@"invalid_token"].location != NSNotFound || [authenticateHeader rangeOfString:@"invalid_request"].location != NSNotFound)) {
```

Comment:
This feels like a server problem maybe. What this does it checks if the request was rejected as unauthenticated and refreshes the token if that is the case. If you change it the way you did auto-token refreshing is broken (which you might not notice if your access tokens do not expire). 

If this change fixed it maybe your server sends 401 in an unexpected way (note that 401 responses without `expired_token` should be handled by your change below, so maybe this line need not be changed?